__tests__/models.test.js

```diff
@@ -209,3 +209,96 @@ describe("GET /api/reviews/:review_id/comments", () => {
     });
   });
 });
+
+describe("PATCH /api/reviews/:review_id", () => {
+  test("200: returns the updated review", () => {
+    return request(app)
+      .patch("/api/reviews/2")
+      .send({ inc_votes: 1 })
+      .expect(200)
+      .then(({ body: { updatedReview } }) => {
+        expect(updatedReview).toHaveProperty("review_id", 2);
+        expect(updatedReview).toHaveProperty("title", "Jenga");
+        expect(updatedReview).toHaveProperty("designer", "Leslie Scott");
+        expect(updatedReview).toHaveProperty("owner", "philippaclaire9");
+        expect(updatedReview).toHaveProperty(
+          "review_img_url",
+          "https://images.pexels.com/photos/4473494/pexels-photo-4473494.jpeg?w=700&h=700"
+        );
+        expect(updatedReview).toHaveProperty(
+          "review_body",
+          "Fiddly fun for all the family"
+        );
+        expect(updatedReview).toHaveProperty(
+          "created_at",
+          "2021-01-18T10:01:41.251Z"
+        );
+        expect(updatedReview).toHaveProperty("votes", 6);
+      });
+  });
+  test("200: database is updated", () => {
+    return request(app)
+      .patch("/api/reviews/2")
+      .send({ inc_votes: -5 })
+      .expect(200)
+      .then(() => {
+        return request(app)
+          .get("/api/reviews/2")
+          .expect(200)
+          .then(({ body: { review } }) => {
+            expect(review).toHaveProperty("review_id", 2);
+            expect(review).toHaveProperty("votes", 0);
+          });
+      });
+  });
+  describe("Error Handlers", () => {
+    test("400: incorrect data type, vote increase is not a number", () => {
+      return request(app)
+        .patch("/api/reviews/2")
+        .send({ inc_votes: "five" })
+        .expect(400)
+        .then(({ body: { msg } }) => {
+          expect(msg).toBe("Bad Request");
+        });
+    });
+    test("400: malformed body", () => {
+      return request(app)
+        .patch("/api/reviews/2")
+        .send({ inc_votes: "null" })
+        .expect(400)
+        .then(({ body: { msg } }) => {
+          expect(msg).toBe("Bad Request");
+        });
+    });
+    //is using psql error 23502 sufficient to test for this?
+
+    test("400: only have inc_votes", () => {
+      return request(app)
+        .patch("/api/reviews/2")
+        .send({ inc_votes: 1, owner: "it's me now" })
+        .expect(400)
+        .then(({ body: { msg } }) => {
+          expect(msg).toBe("Bad Request");
+        });
+    });
+    // when I do this test, the new code I write makes the 23502 psql error handler redundant, if there a situation where we would be able to put this in, or should the code written should be good enought to never leave a null data value?
```

Comment:
I don't think it makes the error handler redundant as you are looking at two different things - the test on 275 is about having an extra property rather than a null value on a never null property.